### PR TITLE
Remove deprecated library

### PR DESCRIPTION
**Pull request by IvanBorislavovDimitrov**

### CODE REVIEW

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildApplicationDeployModelStep.java [hunk 1]
```diff
@@ -87,6 +87,7 @@ private Staging modifyHealthCheckType(Staging staging) {
 
     private List<String> getHealthCheckValues() {
         return Arrays.stream(HealthCheckType.values())
+                     .filter(healthCheckType -> healthCheckType != HealthCheckType.NONE)
```
Comment (radito3) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildApplicationDeployModelStep.java:90]: What does this have to do with the tokens? :D

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java [hunk 2]
```diff
@@ -25,19 +29,24 @@ public TokenParserChain(List<TokenParser> tokenParsers) {
...
-        }
-        return null;
+        OAuth2AccessTokenWithAdditionalInfo parsedToken = parseTokenString(tokenString);
+        logTokenInfo(parsedToken.getOAuth2AccessToken());
+        return parsedToken;
+    }
+
+    private OAuth2AccessTokenWithAdditionalInfo parseTokenString(String tokenString) {
+        return tokenParsers.stream()
+                           .map(tokenParser -> tokenParser.parse(tokenString))
```
[the remark this answers is not in the document]
Comment (radoslav-d) [reply multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java:39]: actually not the same

**Review by radoslav-d — APPROVED** [review]
Good usage of design patterns! Technical difficulty is handled well.

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (radito3) [reply multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java:39]: Oops, my bad :D

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java [hunk 3]
```diff
@@ -25,19 +29,24 @@ public TokenParserChain(List<TokenParser> tokenParsers) {
...
+                           .filter(Optional::isPresent)
+                           .map(Optional::get)
+                           .findFirst()
```
[the remark this answers is not in the document]
Comment (IvanBorislavovDimitrov) [reply multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsers/TokenParserChain.java:42]: Currently, there is only one parser so there will be always one token. And overall one token should be parsed only by one parser if there were more.

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsing/TokenValidationStrategyFactory.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.controller.core.Messages;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+
+@Named
+public class TokenValidationStrategyFactory {
+
+    public ValidationStrategy createStrategy(String algorithm) {
+        switch (algorithm) {
```
Comment (radoslav-d) [inline multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsing/TokenValidationStrategyFactory.java:14]: yeah, a lot easier to read

File: multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsing/RSAValidationStrategy.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        KeyFactory rsaKeyFactory = getRSAKeyFactory();
+        X509EncodedKeySpec x509EncodedKeySpec = new X509EncodedKeySpec(encodedPublicKeyValue);
+        RSAPublicKey rsaPublicKey = getRSAPublicKey(rsaKeyFactory, x509EncodedKeySpec);
+        return new RSASSAVerifier(rsaPublicKey);
+    }
+
+    private String removeX509Wrapper(String publicKey) {
+        return publicKey.replace("-----BEGIN PUBLIC KEY-----", "")
+                        .replace("-----END PUBLIC KEY-----", "")
+                        .replace("\n", "");
```
[the remark this answers is not in the document]
Comment (IvanBorislavovDimitrov) [reply multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/security/token/parsing/RSAValidationStrategy.java:33]: Cannot represent empty char

**Review by radoslav-d — APPROVED** [review]

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildApplicationDeployModelStep.java (hunk 1 above)
Comment (IvanBorislavovDimitrov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildApplicationDeployModelStep.java:90]: Nothing at all and respectively is in another commit

**Review by radito3 — APPROVED** [review]
